cocotb/regression.py

```diff
@@ -714,7 +723,19 @@ def generate_tests(self, prefix="", postfix=""):
             name = "%s%s%s_%03d" % (prefix, self.name, postfix, index + 1)
             doc = "Automatically generated test\n\n"
 
+            # preprocess testoptions to split tuples
+            testoptions_split = {}
             for optname, optvalue in testoptions.items():
+                if ',' in optname:
+                    optname = optname.split(',')
+                if isinstance(optname, tuple) or isinstance(optname, list):
+                    assert len(optname) == len(optvalue)
```

Comment:
This should be an `if` and `raise`, this isn't an appropriate use of `assert`. Also, this check should be moved to `add_option`. Catch errors as early as you can.

Reply:
OK, what should I raise?

Reply:
`ValueError` is kind of a catch all for bad argument values.

Reply:
So, as I understand it, this `assert` is trying to verify a precondition we already ensure in `add_option`? If so, a comment to that effect would be good.

Reply:
I added the check in `add_option` that raises the exception, but kept the assert to ensure no funny business takes place between calling `add_option` and `generate_tests`.  But sure, adding a comment is a good idea.